src/cowboy_websocket.erl

```diff
@@ -411,3 +412,10 @@ handler_terminate(#state{env=Env, handler=Handler},
 		Req, HandlerState, Reason) ->
 	cowboy_handler:terminate(Reason, Req, HandlerState, Handler),
 	{ok, Req, [{result, closed}|Env]}.
+	
+	
+%% @private
+get_real_socket({proxy_socket, _, Socket, _, _, _, _, _, _}) -> Socket;
```

Comment:
expanding `messages/1` would probably work fine for us, it just makes a progressive deployment difficult if the support isn't baked in ahead of time. It's workable, though.